spec/lib/bookmark_manager_spec.rb

```diff
@@ -334,4 +379,127 @@ def notifications_for_user
       end
     end
   end
+
+  describe "#create_for (use_polymorphic_bookmarks)" do
+    before do
+      SiteSetting.use_polymorphic_bookmarks = true
+    end
+
+    it "allows creating a bookmark for the topic and for the first post" do
+      subject.create_for(bookmarkable_id: post.topic_id, bookmarkable_type: "Topic", name: name)
+      bookmark = Bookmark.find_by(user: user, bookmarkable: post.topic)
+      expect(bookmark).not_to eq(nil)
+
+      subject.create_for(bookmarkable_id: post.id, bookmarkable_type: "Post", name: name)
+      bookmark = Bookmark.find_by(user: user, bookmarkable: post)
+      expect(bookmark).not_to eq(nil)
+    end
+
+    it "when topic is deleted it raises invalid access from guardian check" do
```

Comment:
Minor but I think guardian is an internal implementation detail here which we do not have to mention.